dev/TEAL.md

```diff
@@ -531,30 +647,57 @@ pop
+
+In addition, applications must be version 6 or greater to be eligible
+for calling in an inner transaction.
 
 ## Varuint
 
 A '[proto-buf style variable length unsigned int](https://developers.google.com/protocol-buffers/docs/encoding#varint)' is encoded with 7 data bits per byte and the high bit is 1 if there is a following byte and 0 for the last byte. The lowest order 7 bits are in the first byte, followed by successively higher groups of 7 bits.
 
-# What TEAL Cannot Do
-
-Design and implementation limitations to be aware of with various versions of TEAL.
-
-* Stateless TEAL cannot lookup balances of Algos or other assets. (Standard transaction accounting will apply after TEAL has run and authorized a transaction. A TEAL-approved transaction could still be invalid by other accounting rules just as a standard signed transaction could be invalid. e.g. I can't give away money I don't have.)
-* TEAL cannot access information in previous blocks. TEAL cannot access most information in other transactions in the current block. (TEAL can access fields of the transaction it is attached to and the transactions in an atomic transaction group.)
-* TEAL cannot know exactly what round the current transaction will commit in (but it is somewhere in FirstValid through LastValid).
-* TEAL cannot know exactly what time its transaction is committed.
-* TEAL cannot loop prior to v4. In v3 and prior, the branch instructions `bnz` "branch if not zero", `bz` "branch if zero" and `b` "branch" can only branch forward so as to skip some code.
-* Until v4, TEAL had no notion of subroutines (and therefore no recursion). As of v4, use `callsub` and `retsub`.
-* TEAL cannot make indirect jumps. `b`, `bz`, `bnz`, and `callsub` jump to an immediately specified address, and `retsub` jumps to the address currently on the top of the call stack, which is manipulated only by previous calls to `callsub`.
+# What AVM Programs Cannot Do
+
+Design and implementation limitations to be aware of with various versions.
+
+* Stateless programs cannot lookup balances of Algos or other
+  assets. (Standard transaction accounting will apply after the Smart
+  Signature has authorized a transaction. A transaction could still be
+  invalid by other accounting rules just as a standard signed
+  transaction could be invalid. e.g. I can't give away money I don't
+  have.)
+* Programs cannot access information in previous blocks. Programs
+  cannot access information in other transactions in the current
+  block, unless they are a part of the same atomic transaction group.
+* Smart Signatures cannot know exactly what round the current transaction
+  will commit in (but it is somewhere in FirstValid through
+  LastValid).
+* Programs cannot know exactly what time its transaction is committed.
```

Comment:
can we say they (stateful) can know what time the last block finished?

Reply:
This section is about what they _can't_ do.  I think that `LastestTimestamp` docs should explain enough about what they _can_ do.